openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java

```diff
@@ -311,102 +325,102 @@ public KubernetesListMixedOperation lists() {
 
   @Override
   public MixedOperation<Build, BuildList, DoneableBuild, BuildResource<Build, DoneableBuild, String, LogWatch>> builds() {
-    return new BuildOperationsImpl(httpClient, OpenShiftConfig.wrap(getConfiguration()), getNamespace());
+    return new BuildOperationsImpl(httpClient, OpenShiftConfig.withApiGroup(this, BUILD, OpenShiftConfig.wrap(getConfiguration())), getNamespace());
```

Comment:
1. All operations can either get instantiated by the client, or by a ResourceHandler. So, adding handling of the api groups here only covers half of the cases. 

So I think that this needs to move inside the Operation impl itself.

2. All operations have a getConfiguration() method and in some edge cases we do use it. So, this change can potential break things (because it passes to the operation a config with custom urls).

If we move the login inside the operation itself and keep the config object clean, I think we should be fine.